src/implementations/vanilla/components/banner/banner.js

```diff
@@ -0,0 +1,172 @@
+import { queryOne } from '@ecl/dom-utils';
+
+/**
+ * @param {HTMLElement} element DOM element for component instantiation and scope
+ * @param {Object} options
+ * @param {String} options.bannerContainer Selector for the banner content
+ * @param {String} options.bannerVPadding Optional additional padding
+ * @param {String} options.bannerImg Selector for the banner img
+ * @param {String} options.defaultRatio Set the default aspect ratio
+ * @param {String} options.breakpoint Breakpoint from which the script starts operating
+ * @param {Boolean} options.attachResizeListener Whether to attach a listener on resize
+ */
+export class Banner {
+  /**
+   * @static
+   * Shorthand for instance creation and initialisation.
+   *
+   * @param {HTMLElement} root DOM element for component instantiation and scope
+   *
+   * @return {Banner} An instance of Banner.
+   */
+  static autoInit(root, { BANNER: defaultOptions = {} } = {}) {
+    const banner = new Banner(root, defaultOptions);
+    banner.init();
+    root.ECLBanner = banner;
+    return banner;
+  }
+
+  constructor(
+    element,
+    {
+      bannerContainer = '.ecl-banner__container',
```

Comment:
I think that we should use the same kind of selector as in the other components (data-ecl-something).
That way we keep a better independence between the js and css